[New post] What is SageMaker? (#930)
build-on-aws/content: concepts/what-is-sagemaker/index.md

> ---

Reviewer: Good job on this blog. It has clean structure and good intro highlighting how the productive fits into ML. My edits focus primarily on AP Style corrections. Please see my notes below. 

Overall, great job on this blog!

Author: Thank you for the review! I have made the proposed changes!

title: "What is Amazon SageMaker?"

> description: "Leveraging Amazon SageMaker for Scalable and Robust ML Workflows"

Reviewer: ```suggestion
description: "Leverage Amazon SageMaker for scalable and robust ML workflows."
```

## What is Amazon Sagemaker?

Amazon SageMaker is a cloud-based machine learning service that aims to make building, training, and deploying machine learning models more accessible. With SageMaker, developers can upload training data, select a machine learning algorithm, and have models train automatically on managed infrastructure. This removes the need for manually setting up servers and clusters. Once a model is trained, developers can deploy it with one click to start making real-time predictions.

SageMaker provides end-to-end capabilities for a typical machine learning workflow. This includes data preparation, feature engineering, model training, deployment, and monitoring. By handling much of the underlying infrastructure, SageMaker simplifies the process of developing machine learning models.

Companies are using SageMaker for various machine learning applications such as forecasting, recommendation engines, fraud detection and Generative AI. The service continues to evolve by adding new algorithms, instance types, and integrations. Overall, SageMaker aims to reduce the complexity faced by developers when building impactful machine learning models.

> ## Key Capabilities and features of Amazon Sagemaker

Reviewer 2: fix spelling Sagemaker -> SageMaker

> ## Key Capabilities and features of Amazon SageMaker

Reviewer: ```suggestion
## Key Capabilities and Features of Amazon SageMaker
```

> ### Managing machine learning workflows

Reviewer: ```suggestion
### Managing Machine Learning Workflows
```

SageMaker Studio provides an integrated web environment for end-to-end ML workflows. This will let you to easily setup projects, track experiments, organize models and data, visualize results, and collaborate. This allows you to use popular programming languages such as Python and R in a notebook interface very similar to Jupyter Lab.

![SageMaker Studio](images/studio.png "SageMaker Studio for end-to-end ML workflows")

> ### Building models

Reviewer: ```suggestion
### Building Models
```

> SageMaker makes model building easier by providing pre-built containers and algorithms. You provide an algorithm, training code, and dataset, and train it on the provided infrastructure. In addition sagemaker supports popular frameworks like TensorFlow, PyTorch, and XGBoost. You can also bring your own frameworks using custom containers. This flexibility allows rapid testing of different approaches.

Reviewer 2: fix spelling sagemaker -> SageMaker

> ### Training models

Reviewer: ```suggestion
### Training Models
```

> SageMaker offers a fully-managed machine learning platform to help engineers and data scientists quickly build, train, and deploy high-quality models at any scale. You need to specify the compute resources you need like GPU type and number of instances. By using features such as [GPU instances and training compilers](https://aws.amazon.com/sagemaker/faqs), training time and costs can be significantly reduced.

Reviewer: ```suggestion
SageMaker offers a fully managed machine learning platform to help engineers and data scientists quickly build, train, and deploy high quality models at any scale. You need to specify the compute resources you need like GPU type and number of instances. By using features such as [GPU instances and training compilers](https://aws.amazon.com/sagemaker/faqs), training time and costs can be significantly reduced.
```

> ### Deploying models

Reviewer: ```suggestion
### Deploying Models
```

> You can use sagemaker to easily deploy machine learning models for real-time or batch predictions. Once your model is trained in SageMaker, it can be deployed to create an endpoint. This endpoint is a REST API that receives prediction requests, runs the request through the deployed model, and returns predictions.

Reviewer 2: fix spelling sagemaker -> SageMaker

> You can use sagemaker to easily deploy machine learning models for real-time or batch predictions. Once your model is trained in SageMaker, it can be deployed to create an endpoint. This endpoint is a REST API that receives prediction requests, runs the request through the deployed model, and returns predictions.

Reviewer: ```suggestion
You can use SageMaker to easily deploy machine learning models for real time or batch predictions. Once your model is trained in SageMaker, it can be deployed to create an endpoint. This endpoint is a REST API that receives prediction requests, runs the request through the deployed model, and returns predictions.
```

> ### Monitoring models

Reviewer: ```suggestion
### Monitoring Models
```

SageMaker provides tools to monitor the health and performance of machine learning models deployed as endpoints. Developers can enable continuous monitoring on a model endpoint which will track key performance metrics like invocation count, latency, and memory usage. This allows detecting when a model endpoint is unhealthy or performing poorly.

### SageMaker Canvas

> Sagemaker Canvas is a GUI interface that allows you to quickly build, train and deploy machine learning models without writing any code. With Sagemaker Canvas you can drag and drop prebuilt components like datasets, algorithms, and training configurations onto canvas to assemble a full machine learning pipeline.

Reviewer 2: fix spellings Sagemaker -> SageMaker

![SageMaker Canvas](images/canvas.png)

> ### Sagemaker Clarify

Reviewer 2: fix spelling Sagemaker -> SageMaker

> Sagemaker Clarify lets you detect bias in models and explains predictions by providing metrics to identify biases against certain groups in training data and models. you can get individual predictions, which will show you which input features most influenced a specific prediction, which will help in debugging the behaviour of the model.

Reviewer 2: fix spelling Sagemaker -> SageMaker

![SageMaker Clarify](images/clarify.png)

> ### Sagemaker Data Wrangler

Reviewer 2: fix spelling Sagemaker -> SageMaker

> SageMaker Clarify lets you detect bias in models and explains predictions by providing metrics to identify biases against certain groups in training data and models. you can get individual predictions, which will show you which input features most influenced a specific prediction, which will help in debugging the behaviour of the model.

Reviewer: ```suggestion
SageMaker Clarify lets you detect bias in models and explains predictions by providing metrics to identify biases against certain groups in training data and models. You can get individual predictions, which will show you which input features most influenced a specific prediction, which will help in debugging the behaviour of the model.
```

> You can use this feature to quickly prepare and clean data for machine learning without writing any code. you can also use it to combine datasets, transform data, handle missing values, and perform other data preparation and feature engineering tasks to get data ready for training machine learning models.

Reviewer: ```suggestion
You can use this feature to quickly prepare and clean data for machine learning without writing any code. You can also use it to combine datasets, transform data, handle missing values, and perform other data preparation and feature engineering tasks to get data ready for training machine learning models.
```

![SageMaker Data Wrangler](images/datawrangler.png)

### SageMaker Jumpstart

> Amazon SageMaker JumpStart gives you quick access to prebuilt machine learning solutions, sample notebooks, and hosted models that can be customized and deployed with little machine learning expertise required. JumpStart allows you to choose from a catalog of solutions like tabular data processing, computer vision, and natural language processing to get started quickly on building and training ML models. Also JumpStart integrates with other SageMaker capabilities. The notebooks leverage features like Clarify, Data Wrangler, Feature Store, and Model Monitor. You can use sagemake jumpstart to acceleratr your machine learning journey through reusable assets, templates, and models.

Reviewer: ```suggestion
Amazon SageMaker JumpStart gives you quick access to prebuilt machine learning solutions, sample notebooks, and hosted models that can be customized and deployed with little machine learning expertise required. JumpStart allows you to choose from a catalog of solutions like tabular data processing, computer vision, and natural language processing to get started quickly on building and training ML models. Also JumpStart integrates with other SageMaker capabilities. The notebooks leverage features like Clarify, Data Wrangler, Feature Store, and Model Monitor. You can use SageMaker JumpStart to accelerate your machine learning journey through reusable assets, templates, and models.
```

### SageMaker Ground Truth

> You can use Sagemaker Ground Truth to create highly accurate training datasets. Ground Truth helps you with data labeling tasks at sacle. And ground truth provides you with tools to verify, analyze and monitor labeling quality. The integration with S3 also gives you the ability to store your dataset which can later be used to train models with sagemaker training jobs.

Reviewer 2: fix spelling Sagemaker -> SageMaker
fix spelling sagemaker -> SageMaker

![SageMaker Ground Truth](images/groundtruth.png)

### Machine Learning Governance

> You can use sagemaker ML Governance to monitor the entire process of your machine learning models from training to deployment. This give you the ability to deploy responsible models by monitoring the model, detecting biases and more to meet regulatory requirements and increase trust in AI systems.

Reviewer 2: fix spelling sagemaker -> SageMaker

> ### Deploying machine learning models at scale

Reviewer: ```suggestion
### Deploying Machine Learning Models at Scale
```

> Sgaemaker MLOps gives you the abilities to deploy ML models at scale by providing CI/CD pipelines for model building and deployment, model monitoring, automated drift detection and retraining, integration with Git/GitHub for version control and collaboration, and tools for experiment tracking and model lineage. With these capablities you can operationalize ML models into production, manage large volume of ML expirements, and more to improve efficiency and reduce errors while improving the productivity of teams.

Reviewer 2: fix spelling Sgaemaker -> SageMaker

## Use Cases

### Generative AI

Amazon SageMaker JumpStart provides quick access to prebuilt machine learning models that can be deployed on SageMaker with minimal setup. This can be very useful for generative AI models like text generators, image generators, etc. With JumpStart, pre-trained large language models like GPT-3 can be quickly deployed for text generation without having to train the models from scratch. Similarly, generative adversarial networks and variational autoencoders for image generation can be deployed as SageMaker endpoints. By using the prebuilt containers and configurations, developers can avoid the complex process of setting up the infrastructure and environment for training and hosting generative models. Overall, SageMaker JumpStart lowers the barriers to getting started with deploying generative models by providing quick access to popular generative models like the Stability AI Stable Diffusion, AI21 Jurassic and HuggingFace Falcon models. This accelerates the development cycle and allows focusing on model evaluation and improvement rather than environment setup.

> ### Fraud detection

Reviewer: ```suggestion
### Fraud Detection
```

> Fraud is a major challenge that many companies face today, including [Mastercard](https://aws.amazon.com/solutions/case-studies/mastercard-ai-ml-testimonial). With the rise of online transactions and digital payments, fraudsters are constantly finding new ways to game systems and commit fraud. Companies are leveraging Amazon SageMaker to build highly accurate fraud detection models in order to counteract this. These models are trained by using anomaly detection, a principal that determines whether a certain transaction is unusual, relative to the rest. These models usually come with their own set of challenges, such as the imbalanced data problem, which happens because fraudulent transactions occur much less frequently than non-fradulent transactions, thereby resulting in an imbalanced data set. This results in a bias in the model, which makes it less likely to detect fraud since it doesn't know enough about what fraud could look like. To mitigate such challenges, sagemaker also comes with managed (and no code) data preprocessing solutions such as [SageMaker Data Wrangler](https://aws.amazon.com/sagemaker/data-wrangler).

Reviewer 2: fix spelling sagemaker -> SageMaker

### Forecasting

> It is possible to leverage Amazon SageMaker to build forecasting models that predict future sales, demand, and other metrics. Forecasting takes past data and uses it to predict trends that may reoccur in the future. SageMaker provides built-in algorithms like DeepAR that are optimized for time-series forecasting. By bringing your historical time-series data into SageMaker notebooks, you can train forecasting models that learn patterns and seasonality in the data. These models can then be deployed into production to generate forecasts on an ongoing basis.

Reviewer: ```suggestion
It is possible to leverage Amazon SageMaker to build forecasting models that predict future sales, demand, and other metrics. Forecasting takes past data and uses it to predict trends that may reoccur in the future. SageMaker provides built-in algorithms like DeepAR that are optimized for time series forecasting. By bringing your historical time series data into SageMaker notebooks, you can train forecasting models that learn patterns and seasonality in the data. These models can then be deployed into production to generate forecasts on an ongoing basis.
```
